sources/packages/backend/workflow/test/test-utils/mock/mock.utils.ts

```diff
@@ -0,0 +1,172 @@
+ * messages to unblock the workflow.
+ * @param serviceTaskId workflow service task id.
+ * @param options mock creation options.
+ * - `jobCompleteMock` the object to be returned when the job is completed.
+ * If not provided a job complete message will be returned either way.
+ * - `jobMessageMocks` optional messages to be sent to the workflow 'message intermediate catch event'.
+ * For instance, when the assessment-gateway detects that the partner information is required and it
+ * needs to be informed when the partner provided the information.
+ * - `subprocesses` one or more workflow subprocess that contains the service
+ * task id being mocked. For instance, the 'create-income-request-task' can be execute
+ * multiple times (student, partner, parents) but it will happen under different
+ * subprocesses.
+ * Multiple parameters here would indicate the hierarchy of the subprocesses
+ * till the service task id is found (e.g. 'create-income-request-task'). For instance,
+ * if the service task id is a child of the SubprocessB and SubprocessB is a child of
+ * SubprocessA, this array needs to be populated as [SubprocessA, SubprocessB], which will
+ * create the service task unique identifier as SubprocessA_SubprocessB_create_income_request_task.
+ * The most common scenario will be one subprocess, for instance, for 'create-income-request-task' under
+ * the assessment-gateway workflow:
+ * - for student income verification it would be studentIncomeVerificationSubprocess_create_income_request_task
+ * - for partner income verification it would be partnerIncomeVerificationSubprocess_create_income_request_task
+ * @returns mocked objects to be used by the worker.
+ * @example
+ * "create_supporting_users_for_parents_task_result": {
+            "createdSupportingUsersIds": [
+                2000,
+                2001
+            ]
+        },
+    "create_supporting_users_for_parents_task_messageResult": [
+        {
+            "name": "supporting-user-info-received",
+            "correlationKey": "2000",
+            "variables": {},
+            "timeToLive": {
+                "type": "SECONDS",
+                "value": 5,
+                "valueType": "TYPED_DURATION",
+                "unit": "s"
+            }
+        },
+        {
+            "name": "supporting-user-info-received",
+            "correlationKey": "2001",
+            "variables": {},
+            "timeToLive": {
+                "type": "SECONDS",
+                "value": 5,
+                "valueType": "TYPED_DURATION",
+                "unit": "s"
+            }
+        }
+    ],
+ */
+export function createMockedWorkerResult(
+  serviceTaskId: WorkflowServiceTasks,
+  options: {
+    jobCompleteMock?: unknown;
+    jobMessageMocks?: PublishMessageRequest<unknown>[];
+    subprocesses?: WorkflowSubprocesses[];
+  },
+): Record<string, unknown> {
+  let fullServiceTaskId = serviceTaskId.toString();
```

Comment:
for my knowledge, why string conversation?

Reply:
`fullServiceTaskId` should be a string (`serviceTaskId` is an enum) and it could be reassigned at line 158. The final intention is to use `fullServiceTaskId` (as a string) in lines 164 and 168.